### DISCUSSION TITLE
Consider replacing F-Droid with Droidify on Android 12

### DISCUSSION THREAD

**Discussion by TommyTran732** [Divested-Mobile/DivestOS-Build#58] [Ideas]
The current F-Droid client is quite bad as it still targets API 25 and does not work with the new Android 12 feature supporting seamless update with third party app stores.

I think that replacing the F-Droid client with Droidify when you update to Android 12 would be a great idea, since it removes the need for the privileged extension (Droidify supports seamless updates on Android 12 just fine), and does not target an ancient API.

**Comment by SkewedZeppelin** [comment]
Perhaps, I can consider just for 12 branch.
It'd be nice to see the functionality upstreamed into F-Droid instead.

**Reply by SkewedZeppelin** [reply]
Eh, probably not.
I can't override the [repo](https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Patches/Common/android_vendor_divested/prebuilts/etc/additional_fdroid_repos.xml) list via system.
And it has Izzy's repo enabled by defualt, which has some non fully free apps.

**Reply by TommyTran732** [reply]
Oh :(

Yeah it would be nice if it was actually upstreamed to the F-Droid client, but I kinda wonder if it will ever happen though. They are still trying to support Android KitKat :/

**Comment by Tachi107** [comment]
I disagree. Using a third party F-Droid client would mean depending on both F-Droid and that third party, and what happens if the guy maintaining Droidify stops working on the client? Or if the main F-Droid client gains support for seamless updates? I believe that defaults should be strong and stable, and unfortunately many small OSS apps/projects have neither of those qualities :/
I would personally submit a patch to the main client instead, so that everybody would benefit from the upstreamed work :)

**Reply by TommyTran732** [reply]
Well yes,  it adds yet another party to trust. It is not ideal. That said, the default client sucks really badly right now.

I am not an Android developer so I am not entirely sure about this, but it looks like F-Droid really wants to keep support for older Android devices, and thus they compile it for API level 25. I don't know if they will even be willing to compile it for API level 31 and lose support for those older devices or not.

**Reply by Tachi107** [reply]
I'm not an Android dev either, but isn't it possible to add that feature while keeping compat with old api levels? Like `if (sdk_ver >= 32) do_seamless_install(); else { ... }`

Edit: I'm almost sure this is possible, as the Aurora store has different install mechanisms for different android versions
Edit 2: yeah, I seem to remember that a given Android app has a minimum required api level and a targeted one
Edit 3: yep, f-droid requires at least level 22 but targets 25 (see on [apkmirror](https://www.apkmirror.com/apk/f-droid-limited/f-droid/f-droid-1-14-release/f-droid-1-14-android-apk-download/))

**Reply by SkewedZeppelin** [reply]
The issue is more of limited resources than technical issue.
And F-Droid is a slow moving sloth.
